s3_bucket_example.tf

```diff
@@ -1,5 +1,6 @@
 resource "aws_s3_bucket" "example_x" {
-  bucket        = "Example Bucket for use by Service X"
+  count         = "${replace(replace(var.bucket_name,"/.+/","1"),"/^$/","0")}"
```

Comment:
The reason I hadn't previously migrated this to using count is that we do not necessarily want to create identical buckets with identical configurations and policies. As with ELB, if an environment is to have a bucket for some purpose, it will be a unique bucket with a unique configuration for that purpose. Therefore we wont be generating X number of cookie-cutter buckets.

This is why I had "var.example_bucket_name". It was just a placeholder for where a real bucket might exist in the future. For the purposes of testing we should just delete the code for the example bucket entirely. It's only purpose was to say to anyone picking up this code from scratch "look - this is how and were you might create a unique bucket for this environment"